### ISSUE TITLE
oppposite to .only()

### ISSUE THREAD

**Issue by aleszoulek** [hmarr/mongoengine#96]
Hello,

I'd like to have some opposite function to QS.only() (as django has .filter() and .exclude()). I don't have a problem with creating the patch, but I wanted to discuss it first.

1] Use case:
When I store emails into mongo, I'd like to exclude "Email.attachment" subdocument list always (as default). It seems kind of silly to list all other email fields (Email.objects.only('from', 'date', 'body', 'to', ...)), I'd welcome to have a way to state just Email.objects.exclude('attachments'). Does it make sense to you?

2] API (naming)
I have basically to ideas how the API could look like:
- Emails.objects.exclude('attachments')
- Emails.objects.only('-attachments')
  I think the 1st looks better.

3] Chaining
How it should behave when chaining Email.objects.only('a', 'b').exclude('b', 'c') ?
Should it chain and return 'b' only, or should the last one overrule the first and behave just like simple .exclude()? I'd prefer the second behavior.

Thanks,

Ales

**Comment by superbobry** [comment]
Hello, Ales. I was thinking of implementing the thing myself today, so let me know if you still wish (and can) do this. Generally, I like your suggestions:
1. `QuerySet.exclude(...)` makes more sense to me, since it's the same as in Django's QuerySet and "-field_name" is counterintuitive
2. I think `Email.objects.only('a', 'b').exclude('b', 'c')` should result in an exception, since QuerySet doesn't have field 'c' after the .only() call.

**Comment by aleszoulek** [comment]
Hello,

yep, I'm working on that right now. We had a discussion about that at #mongoengine IRC that resulted in folowing:

1] The fuction will be called "exclude". The only concern that I realised is that Django's exclude() is kind of different - it's opposite to "filter", so it's filtering rows not fields.

2] I'm not sure about the exception. At the moment I've implemented it without raising the exception.

3] only and exclude calls will combine, so there will be a .all_fields() method that will "reset" the field filters

I hope I'll have the patch today, or tomorow ready for review.

A.
